### PR TITLE
fix(lib): fix error when passing path option

### PR DESCRIPTION
**Pull request by phhien203**
fix error "Path ... does not exist." when using schematic with option project or path
add unit tests

### CODE REVIEW

**Comment by phhien203** [comment]
Hi @NetanelBasal,
Please help to review my PR. Thanks.

**Comment by NetanelBasal** [comment]
Thanks for the PR. Do you think you can add Github actions to run the specs?

File: src/utils.ts [hunk 1]
```diff
@@ -1,17 +1,33 @@
...
   getWorkspace,
   buildDefaultPath,
 } from '@schematics/angular/utility/workspace';
+import { parseName } from '@schematics/angular/utility/parse-name';
 
 export type Options = { name: string; path: string; project: string };
 type Entity = 'component' | 'directive' | 'pipe';
 
-export function appendModule(options: Options, type: Entity) {
+export function appendModule(options: any, type: Entity) {
```
Comment (NetanelBasal) [inline src/utils.ts:18]: Why any?

**Review by NetanelBasal — APPROVED** [review]

File: src/utils.ts (hunk 1 above)
Comment (phhien203) [reply src/utils.ts]: I added `type?` in `Options` and removed usage of `any`. Thanks

**Comment by phhien203** [comment]
I added Github action to run CI

File: src/component/schema.json [hunk 2]
```diff
@@ -0,0 +1,137 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
```
Comment (NetanelBasal) [inline src/component/schema.json:2]: Why did you add these? I'm referencing the Angular schemas directly. I don't want to maintain it :)

Comment (phhien203) [reply src/component/schema.json:2]: Well, the trick is that we are referencing the schema by relative path.
- On development environment, the path would be `../node_modules/@schematics/angular/component/schema.json`, otherwise it will throw error like `Error: Path "/Users/hien-pham/projects/@schematics/angular/component/schema.json" does not exist.`.
- Whereas on production, the path would be `../../../@schematics/angular/component/schema.json`
That's why I changed the path to `./component/schema.json` to make it works on both development and production. Do you have any idea for this?

Comment (NetanelBasal) [reply src/component/schema.json:2]: Change it in build time?

Comment (NetanelBasal) [reply src/component/schema.json:2]: I'm wondering why I didn't encounter this issue, but I don't remember what I did :)

Comment (phhien203) [reply src/component/schema.json:2]: Let me try your approach :)

File: src/collection.dev.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (phhien203) [inline src/collection.dev.json:1]: Hi @NetanelBasal ,
I came up with this idea and it works. In dev mode, I used `collection.dev.json`, and in production, I use `collection.json`.
Thanks.

File: src/collection.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "$schema": "../node_modules/@angular-devkit/schematics/collection-schema.json",
+  "$schema": "../../../@angular-devkit/schematics/collection-schema.json",
```
Comment (NetanelBasal) [inline src/collection.json:2]: Can you revert this, please? I don't think it'll work when consuming the library

Comment (NetanelBasal) [reply src/collection.json:2]: I mean, there was a reason to change it? It works fine.

Comment (phhien203) [reply src/collection.json]: I reverted it it to previous version. Thanks.
